timer/time.go

```diff
@@ -0,0 +1,28 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2022 Datadog, Inc.
+
+package timer
+
+import (
+	"time"
+)
+
+// timeCache is a simple cache for time.Now() to hopefully avoid some expensive calls to REALTIME part of time.Now()
+type timeCache struct {
```

Comment:
I would have called this `clock`, really.

Reply:
I'll take it